mmeval/metrics/lvis.py

```diff
@@ -0,0 +1,331 @@
+
+        return metric_result
+
+    def compute_metric(self, results: list) -> Dict[str, Union[float, list]]:
+        """Compute the LVIS metrics.
+
+        Args:
+            results (List[tuple]): A list of tuple. Each tuple is the
+                prediction and ground truth of an image. This list has already
+                been synced across all ranks.
+
+        Returns:
+            dict: The computed metric. The keys are the names of
+            the metrics, and the values are corresponding results.
+        """
+        tmp_dir = None
+        if self.outfile_prefix is None:
+            tmp_dir = tempfile.TemporaryDirectory()
+            outfile_prefix = osp.join(tmp_dir.name, 'results')
+        else:
+            outfile_prefix = self.outfile_prefix
+
+        # split gt and prediction list
+        preds, gts = zip(*results)
+
+        # handle lazy init
+        if len(self.cat_ids) == 0:
+            self.cat_ids = self._lvis_api.get_cat_ids()
+        if len(self.img_ids) == 0:
+            self.img_ids = self._lvis_api.get_img_ids()
+
+        # convert predictions to coco format and dump to json file
+        result_files = self.results2json(preds, outfile_prefix)
+
+        eval_results: OrderedDict = OrderedDict()
+        if self.format_only:
+            self.logger.info('results are saved in '
+                             f'{osp.dirname(outfile_prefix)}')
+            return eval_results
+
+        lvis_gt = self._lvis_api
+
+        for metric in self.metrics:
+            self.logger.info(f'Evaluating {metric}...')
+
+            try:
+                lvis_dt = LVISResults(lvis_gt, result_files[metric])
+            except IndexError:
+                self.logger.warning(
+                    'The testing results of the whole dataset is empty.')
+                break
+
+            iou_type = 'bbox' if metric == 'proposal' else metric
+            lvis_eval = LVISEval(lvis_gt, lvis_dt, iou_type)
+            lvis_eval.params.imgIds = self.img_ids
+            metric_items = self.metric_items
+            if metric == 'proposal':
+                lvis_eval.params.use_cats = 0
+                lvis_eval.params.max_dets = self.proposal_nums
+                lvis_eval.evaluate()
+                lvis_eval.accumulate()
+                lvis_eval.summarize()
+                if metric_items is None:
+                    metric_items = [
+                        f'AR@{self.proposal_nums}',
+                        f'ARs@{self.proposal_nums}',
+                        f'ARm@{self.proposal_nums}',
+                        f'ARl@{self.proposal_nums}'
+                    ]
+                for k, v in lvis_eval.get_results().items():
+                    if k in metric_items:
+                        val = float(f'{float(v):.3f}')
+                        eval_results[k] = val
+
+            else:
+                lvis_eval.evaluate()
+                lvis_eval.accumulate()
+                lvis_eval.summarize()
```

Comment:
Same here

Reply:
lvis_eval.summarize() does not print anything.